src/__tests__/CompareResults/beta/RevisionSelect.test.tsx

```diff
@@ -1,12 +1,216 @@
+import ResultsView from '../../../components/CompareResults/beta/ResultsView';
 import RevisionSelect from '../../../components/CompareResults/beta/RevisionSelect';
+import useProtocolTheme from '../../../theme/protocolTheme';
 import { renderWithRouter } from '../../utils/setupTests';
-import { screen } from '../../utils/test-utils';
+import {
+  fireEvent,
+  renderHook,
+  screen,
+  waitFor,
+  within,
+} from '../../utils/test-utils';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual<typeof import('react-router-dom')>('react-router-dom'),
+  useSearchParams: () => [new URLSearchParams({ fakedata: 'true' })],
+}));
 
 describe('Revision select', () => {
+  const protocolTheme = renderHook(() => useProtocolTheme()).result.current
+    .protocolTheme;
+  const toggleColorMode = renderHook(() => useProtocolTheme()).result.current
+    .toggleColorMode;
+
   it('Should match snapshot', () => {
     renderWithRouter(<RevisionSelect />);
 
     expect(screen.getByTestId('revision-select')).toBeInTheDocument();
     expect(document.body).toMatchSnapshot();
   });
+
+  it("Should change 'All revisions' option to bb6a5e451dac", async () => {
+    renderWithRouter(<RevisionSelect />);
+
+    let selectButton = await waitFor(() => screen.getByRole('button'));
+
+    expect(selectButton).toHaveTextContent('All revisions');
+
+    fireEvent.mouseDown(selectButton);
+
+    const listbox = within(screen.getByRole('listbox'));
+
+    fireEvent.click(listbox.getByText('bb6a5e451dac'));
+
+    selectButton = await waitFor(() => screen.getByRole('button'));
+    expect(selectButton).toHaveTextContent('bb6a5e451dac');
+  });
+
+  it('Should render results for bb6a5e451dac', async () => {
+    const location = {
+      ...window.location,
+      search: '?fakedata=true',
+    };
+    Object.defineProperty(window, 'location', {
+      writable: true,
+      value: location,
+    });
+    const urlParams = new URLSearchParams(window.location.search);
+    const fakedataParam = urlParams.get('fakedata');
+    expect(fakedataParam).toBe('true');
+
+    renderWithRouter(
+      <ResultsView
+        protocolTheme={protocolTheme}
+        toggleColorMode={toggleColorMode}
+      />,
+    );
+
+    let comparisonHeaders = await waitFor(() =>
+      screen.getAllByTestId(/revision-header/),
+    );
```

Comment:
use `await screen.findAllByTestId` directly, you don't need `waitFor`.


```suggestion
    let comparisonHeaders = await screen.findAllByTestId(/revision-header/);
```